### DISCUSSION TITLE
AttributeError: module 'scipy.sparse' has no attribute 'coo_array'

### DISCUSSION THREAD

**Discussion by jogugil** [networkx/networkx#5619] [General]
Hi, first of all, excuse me if this is not where to put the doubt, but I'm lost.
I am using the 'textacy' library to solve a pnl problem. It seems that this library uses networkky, which is the one that throws me the following error:
D:\anaconda\envs\NLP\lib\site-packages\networkx\algorithms\link_analysis\pagerank_alg.py in pagerank_scipy(G, alpha, personalization, max_iter, tol, nstart, weight, dangling)
    467
    468 nodelist = list(G)
--> 469 A = nx.to_scipy_sparse_array(G, nodelist=nodelist, weight=weight, dtype=float)
    470 S = A.sum(axis=1)
    471 S[S != 0] = 1.0 / S[S != 0]

D:\anaconda\envs\NLP\lib\site-packages\networkx\convert_matrix.py in to_scipy_sparse_array(G, nodelist, dtype, weight, format)
    919 r += diag_index
    920 c += diag_index
--> 921 A = sp.sparse.coo_array((d, (r, c)), shape=(nlen, nlen), dtype=dtype)
    922 tries:
    923 return A.asformat(format)

AttributeError: module 'scipy.sparse' has no attribute 'coo_array'

Someone could help me. I am stopped. My spacy version is 3.3. As I saw the 'coo_array' function is only in version 1.8, very far from the current one and which I cannot use. This is solved?

**Comment by rossbar** [comment]
Thanks for the question @jogugil - this is likely due to a versioning issue. NetworkX 2.7 requires scipy>=1.8 for the newer scipy.sparse interface. Therefore, if you're using NetworkX > 2.6 you'll either need to upgrade scipy, or if you need the older scipy for some other reason, then install an older (<=2.6) version of networkx.

**Reply by jogugil** [reply]
My version of spacy is spacy 3.3.0 . It doesn't give me the version of scipy.sparse. And the version of networkxy is networkx 2.7.1. But it keeps giving me the problem

**Reply by rossbar** [reply]
You should be able to see your `scipy` version with something like `conda info`. I'm not a conda user, but I think something like `conda install --upgrade scipy` should work. If the dependency resolution fails then you might have to consider downgrading networkx instead.

**Reply by jogugil** [reply]
This is   my environment:

scipy                     1.7.3            py39h0a974cb_0

networkx                  2.7.1              pyhd3eb1b0_0

python                    3.9.7                h6244533_1

**Reply by MridulS** [reply]
@jogugil please update the scipy version with
```
conda update scipy
```

**Reply by jogugil** [reply]
thanks,,

My jupiter did not update that library due to conflicts with other modules. I had to update and reinstall everything. The version of scipy was 1.7. It didn't happen to me in 1.8. It works now, thanks
